etsd/keys/views.py

```diff
@@ -100,6 +72,31 @@ def form_valid(self, form):
         )
         return HttpResponseRedirect(reverse("home"))
 
+class PublicKeyAcceptRejectFormView(UpdateView):
+    model = models.PublicKey
+    form_class = forms.PublicKeyAcceptRejectForm
+
+    def form_valid(self, form):
+        pubk=self.object
+        if pubk.status == "ACTIVE":
+            activekeys = models.PublicKey.objects.filter(authority=pubk.authority, status="ACTIVE")
+            for key in activekeys:
+                key.status = "INACTIVE"
```

Comment:
Let's add a key.deactivated_on attibute (DateTimeField, blank=True, null=True) and set it when a key is deactivated.